lib/charms/haproxy/v0/haproxy_route.py

```diff
@@ -0,0 +1,1346 @@
+# pylint: disable=too-many-lines
+"""Haproxy-route interface library.
+
+## Getting Started
+
+To get started using the library, you just need to fetch the library using `charmcraft`.
+
+```shell
+cd some-charm
+charmcraft fetch-lib charms.haproxy.v0.haproxy_route
+```
+
+In the `metadata.yaml` of the charm, add the following:
+
+```yaml
+requires:
+    backend:
+        interface: haproxy-route
+        limit: 1
+```
+
+Then, to initialise the library:
+
+```python
+from charms.haproxy.v0.haproxy_route import HaproxyRouteRequirer
+
+class SomeCharm(CharmBase):
+  def __init__(self, *args):
+    # ...
+
+    # There are 2 ways you can use the requirer implementation:
+    # 1. To initialize the requirer with parameters:
+    self.haproxy_route_requirer = HaproxyRouteRequirer(self,
```

Comment:
IMO this list of attributes won't be maintained. Can you put a placeholder instead?